snuba/admin/auth.py

```diff
@@ -68,7 +72,14 @@ def get_iam_roles_from_file(user: AdminUser) -> Sequence[str]:
 def _set_roles(user: AdminUser) -> AdminUser:
     # todo: depending on provider convert user email
     # to subset of DEFAULT_ROLES based on IAM roles
-    iam_roles = get_iam_roles_from_file(user)
+    key = f"roles-{user.email}"
+    iam_roles_str = redis_client.get(key)
+    iam_roles = rapidjson.loads(iam_roles_str) if iam_roles_str else None
+    if not iam_roles:
+        iam_roles = get_iam_roles_from_file(user)
+        redis_client.set(key, rapidjson.dumps(iam_roles))
+        redis_client.expire(key, settings.ADMIN_ROLES_REDIS_TTL)
+
```

Comment:
This implementation tightly couples authentication to caching. If your cache fails to update, you'll fail the request. this should be done in such a way that:

1. The distinction between cache and auth is clear
2. If the caching fails the app still works